schema-loader/src/test/java/com/scalar/db/schemaloader/SchemaLoaderTest.java

```diff
@@ -0,0 +1,293 @@
+package com.scalar.db.schemaloader;
+
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+import com.scalar.db.schemaloader.core.SchemaOperator;
+import com.scalar.db.schemaloader.core.SchemaOperatorFactory;
+import java.nio.file.Path;
+import java.util.Collections;
+import java.util.Properties;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mock;
+import org.mockito.MockedStatic;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+
+public class SchemaLoaderTest {
+  @Mock private SchemaOperator operator;
+  @Mock private Path configFilePath;
+  @Mock private Properties configProperties;
+  @Mock private Path schemaFilePath;
+
+  private AutoCloseable closeable;
+  private MockedStatic<SchemaOperatorFactory> schemaOperatorFactoryMockedStatic;
+
+  @Before
+  public void setUp() {
+    closeable = MockitoAnnotations.openMocks(this);
+    schemaOperatorFactoryMockedStatic = Mockito.mockStatic(SchemaOperatorFactory.class);
+    schemaOperatorFactoryMockedStatic
+        .when(
+            () ->
+                SchemaOperatorFactory.getSchemaOperator(
+                    Mockito.any(Path.class), Mockito.anyBoolean()))
+        .thenReturn(operator);
+    schemaOperatorFactoryMockedStatic
+        .when(
+            () ->
+                SchemaOperatorFactory.getSchemaOperator(
+                    Mockito.any(Properties.class), Mockito.anyBoolean()))
+        .thenReturn(operator);
+  }
+
+  @After
+  public void tearDown() throws Exception {
+    schemaOperatorFactoryMockedStatic.close();
+    closeable.close();
+  }
+
+  @Test
+  public void load_WithProperFilePathsArguments_ShouldCallCreateTables() throws Exception {
+    // Act
+    SchemaLoader.load(configFilePath, schemaFilePath, Collections.emptyMap(), true);
+
+    // Assert
+    verify(operator).createTables(schemaFilePath, Collections.emptyMap());
+    verify(operator).createCoordinatorTable(Collections.emptyMap());
+  }
+
+  @Test
+  public void
+      load_WithProperConfigFilePathAndNullPathSchemaArguments_ShouldJustCallCreateCoordinator()
+          throws Exception {
+    // Act
+    SchemaLoader.load(configFilePath, (Path) null, Collections.emptyMap(), true);
```

Comment:
I don't think it's good that we need a cast when passing a null value. 

One idea is to add a something parameter class like `LoadParameters` and its builder class as follows (it's just a pseudocode):
```java
LoadParameters loadParameters = LoadParameters.newBuilder().configFile("<config file path>").withSchemaFile("<schema file path>").withOptions(options).withCoordinatorTableCreation();

SchemaLoader.load(loadParameters);
```

What do you think?

Reply:
If we change to using the Parameter builder, then we should change the interface as well. But with current overloading methods, it looks easy to use. So for the tests, I just initialize a variable with null value to avoid casting. As I see, this case of creating/deleting coordinator table only using in the command which has the `schemaFilePath` is null